app/src/main/java/tech/ula/utils/DownloadUtility.kt

```diff
@@ -52,7 +55,7 @@ class DownloadUtility(
         downloadDirectory.walkBottomUp()
                 .filter { it.name.contains("UserLAnd-") }
                 .forEach {
-                    val delimitedContents = it.name.split("-")
+                    val delimitedContents = it.name.split("-", limit = 3)
```

Comment:
I think this also needs to be changed in `AssetPreferences#getAsset`.

Reply:
The string for `it` is a bit differently formatted than the input for `DownloadUtility`.  For example `libtermux-auth.so-1545942922`

I'm thinking of doing using `split` with a regular expression to separate the string into two strings based on where the last occurrence of the hyphen is.

Reply:
Regardless, I'm pretty sure something will need to be changed in the `AssetPreferences, as `Asset` lists are cached using a string set of the format '<filename>-<remote timestamp>' and they split on that first hyphen.